File: wtplayer/src/WTSpout.cpp
#include "basetypes.hpp"
#include "WTSpout.hpp"
#include "WTBitmap.hpp"
#include "WTSurfaceShader.hpp"

WTSpout::WTSpout(WT* wt_) :
  WTGroup(wt_) {
}

WTSpout::~WTSpout() {
  if(this->shader)
    this->shader->Release();
  this->shader = NULL;

  if(this->texture)
    this->texture->Release();
  this->texture = NULL;
}

void WTSpout::setTexture(WTBitmap* pBitmap){
  APILOG;
  if(this->texture)
    this->texture->Release();
  this->texture = pBitmap;
  this->texture->AddRef();
}

void WTSpout::setSurfaceShader(WTSurfaceShader* pISurfaceShader){
  APILOG;
  if(this->shader)
    this->shader->Release();
  this->shader = pISurfaceShader;
  this->shader->AddRef();
}

void WTSpout::setRate(float Rate){
  APILOG;
}

void WTSpout::setScaleRange(float minScale,
                            float maxScale,
                            int unitLifePoint){
  APILOG;
}

void WTSpout::setParticlePropertyRange(int property,
                                       float minValue,
                                       float maxValue){
  APILOG;
}

void WTSpout::setExternalForce(float x,
                               float y,
                               float Z){
  APILOG;
}

void WTSpout::setColorRange(int minRed,
                            int minGreen,
                            int minBlue,
                            int minAlpha,
                            int maxRed,
                            int maxGreen,
                            int maxBlue,
                            int maxAlpha,
                            int unitLifePoint){
  APILOG;
}

void WTSpout::setSourcePointRange(float minValue,
                                  float maxValue){
  APILOG;
}

void WTSpout::setMode(int mode){
  APILOG;
}

File: wtplayer/javasrc/wildtangent/webdriver/WTSpout.java
package wildtangent.webdriver;

public interface WTSpout
extends WTGroup {
    public void setSourcePointRange(float var1, float var2);

    public void setRate(float var1);

    public void setMode(int var1);

    public void setExternalForce(float var1, float var2, float var3);

    public void setParticlePropertyRange(int var1, float var2, float var3);

    public void setSurfaceShader(WTSurfaceShader var1);

    public void setScaleRange(float var1, float var2, int var3);

    public void setColorRange(int var1, int var2, int var3, int var4, int var5, int var6, int var7, int var8, int var9);

    public void setTexture(WTBitmap var1);
}

